### ISSUE TITLE
git-branch in auto version always says master

### ISSUE THREAD

**Issue by aaronjensen** [boldpoker/edeliver#74]
If you do:

```bash
mix edeliver build release --branch=whatever --auto-version=git-branch+git-revision
```

Your version number will always be `XXX+master-abcdef` where `abcdef` is the latest revision on branch `whatever`

I'd expect it to be `XXX+whatever-abcdef`

**Comment by bharendt** [comment]
Are you using the current edeliver version (`1.2.5`)? This should actually already be fixed by this merged [pull request](https://github.com/boldpoker/edeliver/pull/60).

**Comment by aaronjensen** [comment]
ah, we were.

**Comment by aaronjensen** [comment]
er actually, yes we are using 1.2.5, and no it didn't seem to be fixed

**Comment by bharendt** [comment]
Could you maybe check whether `git rev-parse HEAD` executed on your build host returns the correct revision which was built and what `git branch --contains <that-revision>` returns?

**Comment by aaronjensen** [comment]
After resolving other issues (edeliver not being in applications, wrong build_at, etc) this appears to be working now. I'm going to close this out.

**Comment by aaronjensen** [comment]
FWIW, this appears to be an issue with having upgraded from an older version of edeliver. If I delete the master branch from our build directory and check out the new branch this doesn't seem to happen again.

**Comment by aaronjensen** [comment]
I'm reopening this, after blowing away my build directory and deploying an upgrade it got it wrong again.

**Comment by bharendt** [comment]
True, if the commit is merged into another branch `git branch --contains` would show both then. I think the best solution would then be to pass the `BRANCH` env to the `mix` task on the build host and read it in the [`Mix.Tasks.Release.Version.get_branch/0`](https://github.com/boldpoker/edeliver/blob/20b75b7d20a99a52dc48c3056b66623f354f4d25/lib/mix/tasks/release_version.ex#L338) function

**Comment by aaronjensen** [comment]
No, I don't think that's what is happening. The branch does not exist in the other branch. When the repo first gets pushed up to the build server it has `master` set to be the actual branch that is being built. That branch is checked out as well. If I check out the actual branch manually it seems to work from then on, but this should be unnecessary.

**Comment by bharendt** [comment]
This should be resolved in edeliver version [`1.2.9`](https://hex.pm/packages/edeliver/1.2.9) which includes the [above commit](https://github.com/boldpoker/edeliver/commit/4de6d7dcb2738dd311f4b7c31033b88eacd07db0).
